core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+    await this.clickOn(categoryDropDown);
+    await this.clickOn(category);
+  }
+
+  /**
+   * This function checks if a category has been selected for the exploration.
+   */
+  async expectSelectedCategoryToBe(expectedCategory: string): Promise<void> {
+    await this.page.waitForSelector('.mat-select-value');
+    const selectedCategory = await this.page.evaluate(() => {
+      return (
+        document.querySelector('.mat-select-value') as HTMLElement
+      ).innerText.trim();
+    });
+    if (selectedCategory === expectedCategory) {
+      showMessage(
+        `The category ${selectedCategory}` + ' is same as expectedCategory.'
+      );
+    } else {
+      throw new Error('Category is not correct.');
+    }
+  }
+
+  /**
+   * This function helps in selecting language from dropdown.
+   */
+  async selectALanguage(language: string): Promise<void> {
+    await this.page.evaluate(() => {
+      window.scrollTo(0, 350);
+    });
+
+    await this.clickOn(languageUpdateBar);
+    await this.clickOn(language);
+  }
+
+  /**
+   *  This function verifies that the selected language is displayed correctly.
+   */
+  async expectSelectedLanguageToBe(expectedLanguage: string): Promise<void> {
+    await this.page.waitForSelector(languageUpdateBar);
+    const languageDropdown = await this.page.$(languageUpdateBar);
+    if (!languageDropdown) {
+      throw new Error('Category dropdown not found.');
+    }
+    await languageDropdown.click();
+
+    const selectedLanguage = await this.page.evaluate(() => {
+      const matOption = document.querySelector(
+        '.mat-option.mat-selected'
+      ) as HTMLElement;
+      if (!matOption) {
+        throw new Error('Selected language option not found.');
+      }
+      const selectedLanguageText = matOption.innerText.trim();
+
+      /**
+       * Extract words within parentheses using regex العربية (Arabic) -> Arabic
+       */
+      const matches = selectedLanguageText.match(/\(([^)]+)\)/);
+      if (matches && matches.length >= 2) {
+        return matches[1].trim();
+      } else {
+        // If no parentheses found, assume the whole text is the language.
+        // For eg : English.
+        return selectedLanguageText.trim();
+      }
+    });
+
+    if (selectedLanguage === expectedLanguage) {
+      showMessage(
+        `The language ${selectedLanguage}` + ' is same as expectedLanguage.'
+      );
+    } else {
+      throw new Error('Language is not correct.');
+    }
+    await this.page.keyboard.press('Enter');
+  }
+
+  /**
+   * This function helps in adding tags.
+   */
+  async addTags(TagNames: string[]): Promise<void> {
+    for (let i = 0; i < TagNames.length; i++) {
+      await this.clickOn(addTags);
+      await this.type(addTags, TagNames[i].toLowerCase());
+      await this.page.keyboard.press('Tab');
+    }
+  }
+
+  /**
+   * This function checks if tags are successfully added.
+   */
+  async expectTagsToBeAdded(expectedTags: string[]): Promise<void> {
```

Comment:
expectTagsToMatch

In general, have expectations check the current state (i.e. what the user is looking at), not check that an action was performed in the past.

Reply:
Actually when we access tags with `const tagElements = Array.from(document.querySelectorAll('mat-chip'));` then they came in lowercase.
For eg: If we have add a tag as `Tag1` then when we access with above logic then it will appear like `tag1`.
It is not necessary that we will add tags always in lowercase so it makes me to use this logic here.
Thanks.